lib/spack/spack/cmd/mark.py

```diff
@@ -98,8 +98,9 @@ def do_mark(specs, explicit):
         specs (list): list of specs to be marked
         explicit (bool): whether to mark specs as explicitly installed
     """
-    for spec in specs:
-        spack.store.STORE.db.update_explicit(spec, explicit)
+    with spack.store.STORE.db.write_transaction():
```

Comment:
this is needed cause the db no longer conditionally takes a write lock after using `mark`.